src/main.cpp

```diff
@@ -59,22 +59,26 @@ EShLanguage find_language(const std::string& name)
         return EShLangFragment;
     else if (ext == ".comp")
         return EShLangCompute;
+	throw std::invalid_argument("Unknown file extension!");
 }
 
 json get_diagnostics(std::string uri, std::string content,
         AppState& appstate)
 {
-    glslang::InitializeProcess();
+    FILE fp_old = *stdout;
+    *stdout = *fopen("/dev/null","w");
```

Comment:
This won't work on Windows. You're trying to temporarily silence stdout. Why? Is glslang not behaving well here and coc.nvim doesn't like that? Maybe we can then find a way to make glslang shut up?

Reply:
I haven't dug into glslang much yet but it seems like it was built to be an independent program rather than library so it has a couple places like https://github.com/KhronosGroup/glslang/blob/master/glslang/MachineIndependent/ShaderLang.cpp#L270 that directly emit to stdout. 

When using the stdio method it cause the json rcp stream to be corrupted with random error output that should probably be redirected to stderr. Would it be more portable if I rebind stdout to stderr? Ideally upstream they would provide a logger for everything and not log when used as a library, or emit to stderr directly.

I was having coc run glslsp | tee ~/stdout.txt to spy on the interaction as it is a little tricky to coerce coc to show all the output itself.

Reply:
That's a real pity. I suppose it can't be helped then. Using glslang, there doesn't appear to be a way to do this properly.